### PR TITLE
spicetify: init

### PR DESCRIPTION
**Pull request by brckd**
Creates a theme for Spotify, when used in combination with [spicetify-nix](https://github.com/Gerg-L/spicetify-nix).
Screenshot from 2024-09-25 22-41-07

### CODE REVIEW

**Comment by brckd** [comment]
Ok, so relying on the existence of `programs.spicetify` indeed did not work. But this can be fixed using the method described in https://discourse.nixos.org/t/how-to-check-is-an-option-available/2919/3.

**Comment by brckd** [comment]
Ok, I took inspiration from https://github.com/danth/stylix/blob/cf8b6e2d4e8aca8ef14b839a906ab5eb98b08561/modules/nixvim/nixvim.nix, since it deals with the same implications of relying on an external module.

File: modules/spicetify/spicetify.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            unpackPhase = "true"; # Don't load any source
+            buildPhase = ''
+              cat << EOF > ./color.ini
+              ${colors}
+              EOF
+            '';
+            installPhase = ''
+              mkdir -p $out
+              cp ./color.ini $out
+            '';
```
Comment (danth) [inline modules/spicetify/spicetify.nix:48]: Could this be simplified using `pkgs.writeTextDir`?

File: modules/spicetify/spicetify.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          };
+          sidebarConfig = false;
```
Comment (trueNAHO) [inline modules/spicetify/spicetify.nix:50]: Why should Stylix disable this option?

Comment (trueNAHO) [inline modules/spicetify/spicetify.nix:50]: Why should Stylix disable this option?

File: modules/spicetify/spicetify.nix (hunk 1 above)
Comment (brckd) [reply modules/spicetify/spicetify.nix:48]: Oh I was hoping there were functions like these when I asked in the NixOS discord server, but got no response 😆. I will look into it!

File: modules/spicetify/spicetify.nix (hunk 2 above)
Comment (brckd) [reply modules/spicetify/spicetify.nix:50]: It probably doesn't matter, I just copied it from the config for the default theme. This option seems to control how the theme can modify spotify besides the color scheme. Will remove it if possible!

Comment (brckd) [reply modules/spicetify/spicetify.nix]: I removed it and it seems to work!

File: modules/spicetify/spicetify.nix (hunk 1 above)
Comment (brckd) [reply modules/spicetify/spicetify.nix]: I ended up using `pkgs.writeTextDir`. `builtins.toFile` doesn't seem to create a directory with a file of the given name, which is requred here.

**Review by trueNAHO — APPROVED** [review]
I didn't test this, but LGTM:

```
Reviewed-by: NAHO <90870942+trueNAHO@users.noreply.github.com>
```

**Review by danth — APPROVED** [review]
LGTM too

**Comment by soulsoiledit** [comment]
Setting `sidebarConfig = false` removes an annoying startup notification about sidebar configuration not being supported with the global navbar.

**Comment by brckd** [comment]
Oh thanks for pointing that out! Since I didn't reinstall spicetify, I didn't see that during my testing. Feel free to create a PR, otherwise I will make one tomorrow.

**Comment by brckd** [comment]
Also make sure to leave a comment in the code, since it's not self explainatory.

**Pull request merged**